src/com/google/cose/utils/CoseUtils.java

```diff
@@ -45,4 +66,93 @@ public static DataItem encodeStructure(String context, Map protectedBodyHeaders,
     }
     return arrayBuilder.end().build().get(0);
   }
+
+  public static LinkedHashMap<Integer, DataItem> getLabelsFromMap(Map keyMap)
+      throws CborException {
+    final LinkedHashMap<Integer, DataItem> labels = new LinkedHashMap<>();
+    for (DataItem item : keyMap.getKeys()) {
+      labels.put(CborUtils.asInteger(item), keyMap.get(item));
+    }
+    return labels;
+  }
+
+  /**
+   * Returns DataItem from a cbor map based on Integer index.
+   * @param cborMap map that has the information.
+   * @param index integer index to be used as key in the map.
+   * @return value in the map corresponding to key
+   */
+  public static DataItem getValueFromMap(Map cborMap, int index) {
```

Comment:
Nit: Index is a bit of a misnomer, since this isn't an array. I think `value` may still be a more appropriate term here.